### ISSUE TITLE
i18n vue plugin option ?

### ISSUE THREAD

**Issue by ArcadeCode** [vuejs/create-vue#545]
### Please describe

I use [vue I18n](https://vue-i18n.intlify.dev/) who is really helpful for Internationalization but he need some configuration and I will really appreciate if I can have an option in create-vue for this, but I checked the code and I'm not qualified to understand it.
### Additional context

To make a fast configuration the vue-i18n doc propose to config all in the main.ts but it's more clear to make a local(s) folder with all traductions and import it in the main.ts or use a i18n folder: 
![image](https://github.com/user-attachments/assets/4f7ab16b-2ad3-48bd-bb3a-eddd804a7bf7)

Anyone think it will be good idea ?

**Comment by Procrustes5** [comment]
I am developing in Japanese and Korean, totally agree with you.

When the project grows to a certain extent, I have often thought that it would be nice if the i18n was set up at the stage of making the project.

I think it's also good for the configuration of creating folders on i18n.

**Comment by Procrustes5** [comment]
https://github.com/vuejs/create-vue/pull/548
Made a pull request based on your issue. If you have any opinions, please give it to me.

**Comment by messenjer** [comment]
In general, I put my i18n configuration into a separate folder: `plugins/i18n.ts`

I keep the translation files in the locales folder: `src/locales/en.json`.

This separation makes it easy to specify the folder containing the translations when using various translation tools:

* The VSCode extension: https://github.com/lokalise/i18n-ally

```
{
    "i18n-ally.localesPaths": [
        "src/locales"
    ]
}
```

* CI/CD : https://lokalise.com/product/apps/code-repositories/github 
* The Vite plugin for vue-i18n: https://github.com/intlify/bundle-tools/blob/main/packages/unplugin-vue-i18n/README.md 

```
export default defineConfig({
  plugins: [
    VueI18nPlugin({
      include: [path.resolve(__dirname, './src/locales/**')],
    }),
  ],
})
```

I find this approach to be more modular.

**Comment by ArcadeCode** [comment]
Yes it will be more modular in this way, but in this idea why don't moving router.ts in the plugins folder ?

**Comment by messenjer** [comment]
In the previous vue installer `vue-cli-i18n` https://github.com/intlify/vue-cli-plugin-i18n 

The file was located here : 

`src/i18n.(js|ts)`

And the default localeDir was `src/locales`

it should be another solution.

**Comment by ArcadeCode** [comment]
Sounds good to me if we take the example from the i18n vue-cli plugin.

**Comment by haoqunjiang** [comment]
`vue-i18n`'s weekly downloads are on par with `pinia` and `@vue/babel-plugin-jsx` (<https://npmtrends.com/@vue/babel-plugin-jsx-vs-pinia-vs-vue-i18n>). So I agree it's definitely worth adding to the scaffold. But I'm not an experienced user of it, any help would be greatly appreciated.
